tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\collabora_online\ExistingSiteJavascript;
+
+use Drupal\file\Entity\File;
+use Drupal\media\Entity\Media;
+use Drupal\media\MediaInterface;
+use WebDriver\Exception\StaleElementReference;
+use weitzman\DrupalTestTraits\ExistingSiteSelenium2DriverTestBase;
+
+/**
+ * Test for Collabora Online editors embedded in or accessed from Drupal.
+ *
+ * To make this a regular FunctionalJavascript test, the WOPI request from
+ * Collabora to Drupal would need to send the SIMPLETEST_USER_AGENT cookie that
+ * tells Drupal that the request should be handled by the test installation,
+ * rather than the regular/existing installation.
+ *
+ * @see \drupal_valid_test_ua()
+ */
+class CollaboraIntegrationTest extends ExistingSiteSelenium2DriverTestBase {
+
+    /**
+     * Tests the Collabora editor in readonly mode.
+     */
+    public function testCollaboraPreview(): void {
+        $user = $this->createUser([
+            'preview document in collabora',
+        ]);
+        $this->drupalLogin($user);
+        $media = $this->createDocumentMedia('Shopping list', 'shopping-list', 'Chocolate, pickles');
+        $this->drupalGet('/cool/view/' . $media->id());
+        $this->getSession()->switchToIFrame('collabora-online-viewer');
+
+        $assert_session = $this->assertSession();
+        $canvas = $assert_session->waitForElement('css', 'canvas#document-canvas');
+        $this->assertNotNull($canvas, 'The canvas element was not found after 10 seconds.');
+
+        $document_field = $assert_session->waitForElement('css', 'input#document-name-input');
+        $this->assertNotNull($document_field, 'The document name input was not found after 10 seconds.');
+        $this->getCurrentPage()->waitFor(10, function() use ($document_field) {
+          return $document_field->getValue() === 'shopping-list.txt';
```

Comment:
```suggestion
          return (string) $document_field->getValue() !== '';
```
We could use this to get the old behavior, where we stop waiting if the string is not empty.

Reply:
I previously thought we need to repeat the selector every time to avoid the stale reference.
But seems like the opposite is the case.